cmd/ocm/account/roles/cmd.go

```diff
@@ -32,10 +32,16 @@ var args struct {
 }
 
 var Cmd = &cobra.Command{
-	Use:   "roles [role-name]",
+	Use:   "roles [flags] [ROLE_NAME]",
 	Short: "Retrieve information of the different roles",
-	Long:  "Get description of a role or list of all roles ",
-	RunE:  run,
+	Long:  "Get description of a role or list of all roles",
+	Args: func(cmd *cobra.Command, args []string) error {
```

Comment:
You could use `cobra.MaximumNArgs(1)` here.

Reply:
For now I figured custom validation lets me have better control of messages.  
I dumped my thoughts on this https://github.com/spf13/cobra/issues/571#issuecomment-707231598

Part of the problem using built-in validation for us is that we disabled printing usage on error (`SilenceUsage: true`).
The default behavior of dumping full description of all args on any error is too annoying for my taste too.  
I think best behavior would be to print just the `Usage:  ocm account roles [flags] [ROLE_NAME]` line on errors (user can always --help for details); if we had that it'd be reasonable to go back to `cobra.MaximumNArgs` and friends.
Opened https://github.com/spf13/cobra/issues/1252, currently we can't `SetUsageTemplate` to something briefer without breaking --help, working on upstream improvement.

Reply:
I did add/leave many cobra.*Args and similar, without reviewing each message, largely because of laziness — this PR actually grew from 1 command with wrong help I wanted to fix :rofl:

Reply:
Read your threads upstream and I agree, if we can expose `UseLine()` when Arg fails validation that would get us very close to full-lazy cmd definitions :)